File: server/java/src/main/java/com/onlinepayments/example/service/dtos/hostedcheckout/CreateHostedCheckoutResponseDto.java
package com.onlinepayments.example.service.dtos.hostedcheckout;

import com.onlinepayments.example.service.domain.common.enums.Currency;

import java.math.BigDecimal;

public record CreateHostedCheckoutResponseDto(
        String hostedCheckoutId,
        String redirectUrl,
        String returnMAC,
        BigDecimal amount,
        Currency currency
) { }

File: server/dotnet/sdk-dotnet-example/Business/DTOs/CreateHostedCheckouts/CreateHostedCheckoutResponseDto.cs
using Business.Domain.Common.Enums;

namespace Business.DTOs.CreateHostedCheckouts;

public class CreateHostedCheckoutResponseDto
{
    public string? HostedCheckoutId { get; init; }
    
    public string? RedirectUrl { get; init; }
    
    public string? ReturnMAC { get; init; }
    
    public decimal? Amount { get; set; }
    
    public Currency? Currency { get; set; }
}